### PR TITLE
Update dependencies on consensus and ledger 

### PR DESCRIPTION
**Pull request by dcoutts**
A couple tracers gained some extra warning fields. Log the new ledger
warnings

Only properly supported for the JSON logging, not for the human
readable, since I can't quite see how it's supposed to be done without
HasTextFormatter constaints propagating everywhere, which does not seem
to be the intended pattern.

### CODE REVIEW

**Comment by dcoutts** [comment]
@CodiePP perhaps you can enlighten me.

**Comment by dcoutts** [comment]
This also depends on a one-liner in consensus
```--- a/ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Ledger.hs
+++ b/ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Ledger.hs
@@ -20,6 +20,7 @@
 module Ouroboros.Consensus.HardFork.Combinator.Ledger (
     HardForkLedgerError(..)
   , HardForkEnvelopeErr(..)
+  , HardForkLedgerWarning(..)
     -- * Type family instances
   , Ticked(..)
     -- * Low-level API (exported for the benefit of testing)
```

File: cabal.project [hunk 1]
```diff
@@ -168,78 +168,78 @@ source-repository-package
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/cardano-ledger-specs
-  tag: 6611345670dad971fbd03594a00b5ef8afa4cae6
-  --sha256: 0rhdlf7kz62231127ywbakhsgsxhhkba5gz24dfy7l9r45a1ijkk
+  tag: 47e2075f751d8370222a2aad32f37699a341ec40
```
Comment (mrBliss) [inline cabal.project:171]: I was planning to update the dependency on cardano-ledger-specs today, now that https://github.com/input-output-hk/cardano-ledger-specs/pull/1664 is merged.

Is that alright or does that perhaps breaking changes that are not wanted?

Comment (mrBliss) [reply cabal.project:171]: @nc6 is working on this already.

Comment (dcoutts) [reply cabal.project:171]: More than one update is fine. Yes we'll need to do it again to get the latest ledger changes.

**Comment by dcoutts** [comment]
bors merge

File: cardano-config/src/Cardano/TracingOrphanInstances/Consensus.hs [hunk 2]
```diff
@@ -321,10 +325,17 @@ instance (Condense (HeaderHash blk), LedgerSupportsProtocol blk)
...
           "Block fits onto some fork: " <> condenseT pt
-        ChainDB.AddedToCurrentChain _ _ c -> \_o ->
+        ChainDB.AddedToCurrentChain _ws _ _ c -> \_o ->
           "Chain extended, new tip: " <> condenseT (AF.headPoint c)
-        ChainDB.SwitchedToAFork _ _ c -> \_o ->
+--        Text.concat [ "\nWarning: " <> formatText w o | w <- ws ]
+--TODO: we could trace these warnings but then a
+-- HasTextFormatter (LedgerWarning blk) constraint will propagate everywhere
+-- and it's not clear that's how it's supposed to work, since we don't have any
+-- other HasTextFormatter constraints floating around
```
Comment (mrBliss) [inline cardano-config/src/Cardano/TracingOrphanInstances/Consensus.hs:334]: I haven't looked at this in detail, but would `show` work for now?

**Comment by dcoutts** [comment]
bors merge

File: cardano-config/src/Cardano/TracingOrphanInstances/Consensus.hs (hunk 2 above)
Comment (dcoutts) [reply cardano-config/src/Cardano/TracingOrphanInstances/Consensus.hs]: done

**Pull request merged**